app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -91,6 +100,16 @@ class BrowserFragment : WebFragment(), CursorEvent {
         })
     }
 
+    override fun onEvent(event: BrowserNavigationOverlay.NavigationEvent, value: String?, autocompleteResult: InlineAutocompleteEditText.AutocompleteResult?) {
+        when (event) {
+            BrowserNavigationOverlay.NavigationEvent.BACK -> if (canGoBack()) goBack()
+            BrowserNavigationOverlay.NavigationEvent.FORWARD -> if (canGoForward()) goForward()
+            BrowserNavigationOverlay.NavigationEvent.RELOAD -> reload()
+            BrowserNavigationOverlay.NavigationEvent.HOME -> (activity as MainActivity).showHomeScreen()
```

Comment:
BrowserFragment should not know about MainActivity - MainActivity should set a listener on BrowserFragment for the necessary events.

If we do it only for small cases, it may not be the end of the world (especially because we're not really testing BF) but it's a nice to have

Reply:
Agreed - this is just a stop-gap for now, I'll update that in a later PR.

Reply:
fwiw, you can add a comment saying it's temporary so I don't nit you about it. :)